src/styles/calendar.scss

```diff
@@ -0,0 +1,81 @@
+$cell-size: 100px;
+$border: 1px solid black;
+$day-color: #eee;
+$hover-color: #ffbfcb;
+$font-size: 30px;
+$animation-duration: 0.5s;
+$days-of-week: (
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+  sun: 6,
+);
+$gap: 1px;
+$font-family: Arial, sans-serif;
+
+html {
+  height: 100%;
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: calc((7 * #{$cell-size}) + 6px);
+  padding: 10px;
+  gap: $gap;
+  counter-reset: day-counter 0;
+
+  &__day {
+    width: $cell-size;
+    height: $cell-size;
+    justify-content: center;
+    align-items: center;
+    font-family: $font-family;
+    font-size: $font-size;
+    background-color: $day-color;
+    border: $border;
+    display: flex;
+    box-sizing: border-box;
+    transition:
+      transform $animation-duration ease,
+      background-color $animation-duration ease;
+
+    &::before {
+      counter-increment: day-counter;
+      content: counter(day-counter);
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: $hover-color;
+      transform: translateY(-20px);
+    }
+  }
+
+  @each $day, $index in $days-of-week {
+    &--start-day-#{$day} .calendar__day:nth-child(1) {
+      margin-left: calc(#{$index} * (#{$cell-size} + 1px));
+    }
+  }
```

Comment:
The `@each` loop here is redundant because a similar loop is defined later starting at line 75. Consider removing one of these loops to avoid duplication and maintain cleaner code.